is_core/filters/default_filters.py

```diff
@@ -440,19 +441,85 @@ class DateTimeFilter(DateFilter):
     extra_suffixes = ['day', 'month', 'year', 'hour', 'minute', 'second']
 
 
+class AbstractDateRangeWidget(widgets.MultiWidget):
+
+    def __init__(self, filter_name):
+        class_from, class_to = self.get_range_classes()
+        super(AbstractDateRangeWidget, self).__init__(
+            (
+                widgets.TextInput(attrs={'class': class_from, 'data-filter': '{}__gte'.format(filter_name)}),
+                widgets.TextInput(attrs={'class': class_to, 'data-filter': '{}__lte'.format(filter_name)}),
+            )
+        )
+
+    def get_range_classes(self):
+        raise NotImplemented
+
+    def decompress(self, value):
+        return []
+
+
+class DateRangeWidget(AbstractDateRangeWidget):
+
+    def get_range_classes(self):
+        return 'date-range-from', 'date-range-to'
+
+
+class DateTimeRangeWidget(AbstractDateRangeWidget):
+
+    def get_range_classes(self):
+        return 'datetime-range-from', 'datetime-range-to'
+
+
+class DateRangeFilterMixin(object):
+
+    def __init__(self, *args, **kwargs):
+        super(DateRangeFilterMixin, self).__init__(*args, **kwargs)
+        self.widget = self.get_widget_class()(self.get_filter_name())
```

Comment:
musi to tak byt protoze init AbstractDateRangeWidget-u prijima filter_name jako pozicni parametr